keras_core/metrics/metric.py

```diff
@@ -72,7 +72,9 @@ def update_state(self, y_true, y_pred, sample_weight=None):
             values = ops.cast(values, self.dtype)
             if sample_weight is not None:
                 sample_weight = ops.cast(sample_weight, self.dtype)
-                sample_weight = ops.broadcast_to(sample_weight, values.shape)
+                sample_weight = ops.broadcast_to(
+                    sample_weight, ops.shape(values)
+                )
```

Comment:
Actually maybe the explicit `broadcast_to` is not necessary, in the next line `ops.multiply` should automatically do the broadcasting. Not 100% sure tho.

Reply:
This is in the docstring of the `metrics`, so it might be better to keep it explicit and might be useful in some cases.

Reply:
kk, sgtm!